upgrade/http.go

```diff
@@ -29,7 +29,9 @@ type httpHandler interface {
 	getHTTP(url string) ([]byte, error)
 }
 
-var httpHandlerKey = struct{}{}
+type httpContextKey struct{}
```

Comment:
Addresses linter warning SA1029: should not use empty anonymous struct as key for value; define your own type to avoid collisions